selene_sdk/samplers/dataloader.py

```diff
@@ -0,0 +1,316 @@
+    For the sequences array, represent unknown bases ("N"s) by binary
+    data with all-ones in the encoding - they will be transformed to
+    the correct representations in selene_sdk.sequences.Genome when unpacked.
+    In addition, to unpack correctly, the length of the packed dimensions,
+    i.e. :math:`L` and :math:`F` must be provided in two integer scalars
+    named `{sequence_key}_length` and `{targets_key}_length` in the HDF5 file
+    if `unpackbits==True`.
+
+    `H5DataLoader` can be used with `MultiSampler` by passing
+    `SamplerDataLoader` object as `train_sampler`, `validate_sampler` or
+    `test_sampler` when initiating a `MultiSampler`.
+
+    Parameters
+    ----------
+    file_path : str
+        The file path of the hdf5 file.
+    in_memory : bool, optional
+        Default is False. If True, load entire dataset into memory.
+    num_workers : int, optional
+        Default is 1. If greater than 1, use multiple processes to parallelize data
+        sampling.
+    use_subset : int, (int, int), list(int), or None, optional
+        Default is None. If a single integer value is provided, sample from only
+        the first `use_subset` rows of the dataset. If a tuple of integers is
+        provided, `(<start-index>, <end-index>)`, sample from the range of
+        rows specified. If a list of integers is provided, restrict to sampling
+        only the indices specified in the list. If None, use the entire dataset.
+    batch_size : int, optional
+        Default is 1. Specify the batch size of the DataLoader.
+    shuffle : bool, optional
+        Default is True. If False, load the data in the original order.
+    unpackbits : bool, optional
+        Default is False. If True, unpack binary-valued array from uint8
+        sequence and targets array. See `numpy.packbits` for details.
+    sequence_key : str, optional
+        Default is "sequences". Specify the name of the hdf5 dataset that contains
+        sequence data.
+    targets_key : str, optional
+        Default is "targets". Specify the name of the hdf5 dataset that contains
+        target data.
+
+    Attributes
+    ----------
+    dataset : `_H5Dataset`
+        The `_H5Dataset` to load data from.
+
+    """
+    def __init__(self,
+                 filepath,
+                 in_memory=False,
+                 num_workers=1,
+                 use_subset=None,
+                 batch_size=1,
+                 shuffle=True,
+                 unpackbits=False,
+                 sequence_key="sequences",
+                 targets_key="targets"):
+        args = {
+            "batch_size": batch_size,
+            "num_workers": 0 if in_memory else num_workers,
+            "pin_memory": True
+        }
+        if use_subset is not None:
+            from torch.utils.data.sampler import SubsetRandomSampler
+            if isinstance(use_subset, int):
+                use_subset = list(range(use_subset))
+            elif isinstance(use_subset, range):
```

Comment:
I changed this in the documentation to a tuple of start and end just because I thought it would be easier for some users to understand - is it OK if we change the code here too to accommodate that?

Reply:
this is changed now